crypto/src/signature.rs

```diff
@@ -343,4 +343,44 @@ pub mod json {
             Ok(s.map(|v| v.0))
         }
     }
+
+    pub mod signature_type {
+        use super::*;
+        use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+        #[derive(Debug, Deserialize, Serialize)]
+        #[serde(rename_all = "lowercase")]
+        enum JsonHelperEnum {
+            Bls,
+            Secp256k1,
+        }
+
+        #[derive(Serialize, Deserialize)]
+        #[serde(transparent)]
+        pub struct SignatureTypeJson(#[serde(with = "self")] pub SignatureType);
+
+        pub fn serialize<S>(m: &SignatureType, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            let json = match m {
+                SignatureType::BLS => JsonHelperEnum::Bls,
+                SignatureType::Secp256k1 => JsonHelperEnum::Secp256k1,
+            };
+            json.serialize(serializer)
+        }
+
+        pub fn deserialize<'de, D>(deserializer: D) -> Result<SignatureType, D::Error>
+        where
+            D: Deserializer<'de>,
+        {
+            let json_enum: JsonHelperEnum = Deserialize::deserialize(deserializer)?;
+
+            let signature_type = match json_enum {
+                JsonHelperEnum::Bls => SignatureType::BLS,
+                JsonHelperEnum::Secp256k1 => SignatureType::Secp256k1,
+            };
+            Ok(signature_type)
+        }
+    }
```

Comment:
Is there really an inconsistency in signature json representations? It is serialized as an integer inside of Signature json (is this wrong and we can just make this the default serialization impl of SignatureType?)

Reply:
Yeah, it is serialized in json as `bls` or `secp256k1`. Ill make it the default.

Reply:
Lol nvm, i just double checked, there is an inconsistancy. In the wallet rpc, they'll serialize the signature type as `bls` or as `secp256k1`, but in the blockheader for example, they'll use 1 or 2... So i think it is fine to leave this as is

Reply:
so weird hahahah

Reply:
Probably one of the lesser weird things xD